### PR TITLE
feat: add dependencies and their licenses

### PR DESCRIPTION
**Pull request by jithine**
## Context

Updating documentation with dependencies

## Objective

Improve documentation

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: docs/user-guide/FAQ.md [hunk 1]
```diff
@@ -142,6 +142,6 @@ If shallow cloning is left enabled and you wish to push back to your git reposit
 
 ## What are the minimum software requirements for a build image?
 
-Screwdriver has no restriction on build container image. However it should have at the minimum `curl` && `openssh` installed.
+Screwdriver has no restriction on build container image. However container should have at the minimum `curl` && `openssh` installed. And default user of the container should be either `root` or have sudo NOPASSWD enabled.
```
Comment (tkyi) [inline docs/user-guide/FAQ.md]: However `the` container

**Comment by tkyi** [comment]
Do we need to add Postgres, sqlite, mssql, mysql, RDB (sequelize) etc and S3?

File: docs/cluster-management/dependencies.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ | Helm (Optional)           | For [deploying Screwdriver into k8s](https://github.com/screwdriver-cd/screwdriver-chart) | [Apache 2.0](https://github.com/helm/helm/blob/master/LICENSE) |
+ | HyperContainer (Optional) | Hypervisor-agnostic Docker runtime. Used with [executor-k8s-vm](https://github.com/screwdriver-cd/executor-k8s-vm) | [Apache 2.0](https://github.com/hyperhq/hyperd/blob/master/LICENSE) |
+ | Jenkins (Optional)     | Used in [executor-jenkins](https://github.com/screwdriver-cd/executor-jenkins)  | [MIT](https://jenkins.io/license) |
+ | JWT                    | For authentication & authorization | [MIT](https://github.com/jsonwebtoken/jsonwebtoken.github.io/blob/master/LICENSE.txt) |
+ | Kubernetes             | For running Screwdriver at scale using [executor-k8s](https://github.com/screwdriver-cd/executor-k8s) | [Apache 2.0](https://github.com/kubernetes/kubernetes/blob/master/LICENSE) |
+ | MariaDB(Optional) | One of the options for Database | [GPL Licence](https://mariadb.com/kb/en/library/licensing-faq/) |
+ | MySQL(Optional) | One of the options for Database | [GPL or Commercial](https://www.mysql.com/about/legal/) |
```
Comment (tkyi) [inline docs/cluster-management/dependencies.md]: wonder if these should all be linked to datastore-sequelize

Comment (jithine) [reply docs/cluster-management/dependencies.md]: updated

**Review by tkyi — APPROVED** [review]
LGTM for now

**Comment by tkyi** [comment]
Jithin can you remove the CLA check for this repo too?

**Pull request merged**